src/Caching/Storages/FileStorage.php

```diff
@@ -143,7 +143,7 @@ public function lock($key)
 	{
 		$cacheFile = $this->getCacheFile($key);
 		if ($this->useDirs && !is_dir($dir = dirname($cacheFile))) {
-			@mkdir($dir); // @ - directory may already exist
+			@mkdir($dir, 0777, true); // @ - directory may already exist
```

Comment:
What about this solution? - http://stackoverflow.com/questions/15356189/creating-a-directory-from-multiple-threads/21626808#21626808